arrow-integrations-jackson-module/src/main/kotlin/arrow/integrations/jackson/module/internal/InjectField.kt

```diff
@@ -0,0 +1,3 @@
+package arrow.integrations.jackson.module.internal
+
+class InjectField<T>(val fieldName: String, val point: (Any?) -> T)
```

Comment:
Nit: Should this be a static nested class of `ProductTypeDeserializer` to give it better namespacing?

Reply:
hmm this is used in both union and product type deserializers, i might need to restructure this a bit. It may make sense to actually use dedicated types for each deserializers. I'll address this

Reply:
awesome i've addressed this. We now use each types namespaced under each serializer / deserializer